etc/dnf/automatic.conf

```diff
@@ -21,6 +21,15 @@ download_updates = yes
 # install.timer override this setting.
 apply_updates = no
 
+# When the system should reboot following upgrades:
+# never                              = don't reboot after upgrades
+# when-changed                       = reboot after any changes
+# when-needed                        = reboot when necessary to apply changes
+reboot = never
+
+# The command that is run to trigger a system reboot.
+reboot_command = "shutdown -r"
```

Comment:
I was testing your code today, and I was thinking whether our ci is suitable or not for testing what you want. Probably, there is no simple way to trigger a reboot, but there are other options.
Since you set a custom command here, I would think of another way to test the reboot trigger.

Maybe, during behave tests, you could set an arbitrary command in `etc/dnf/automatic.conf`. Let's say that instead of `shutdown -r` you `touch` some file, then you can check if the file exists and have the trigger tested. Unfortunately this is the only partial solution I came up with.

Reply:
FTR: Cockpit's updates page [delays the shutdown by a bit](https://github.com/cockpit-project/cockpit/blob/5096b8e26001b6919cc405a093757bfdb7fd033b/pkg/packagekit/autoupdates.jsx#L186), so that logged in users get a warning and have some time to orderly stop their work. That's a good thing to do on servers IMHO.

That same thing also allows you to test this feature relatively painlessly: It's enough time to validate that a shutdown was scheduled and cancel it again ([like in our tests](https://github.com/cockpit-project/cockpit/blob/5096b8e26001b6919cc405a093757bfdb7fd033b/test/verify/check-packagekit#L1667)). That avoids the costly (time) or impossible (Testing Farm/gating tests) actual reboot.

Reply:
> FTR: Cockpit's updates page [delays the shutdown by a bit](https://github.com/cockpit-project/cockpit/blob/5096b8e26001b6919cc405a093757bfdb7fd033b/pkg/packagekit/autoupdates.jsx#L186), so that logged in users get a warning and have some time to orderly stop their work. That's a good thing to do on servers IMHO

That's a good idea.

> That same thing also allows you to test this feature relatively painlessly

That does not apply for `dnf`, since we run our tests in containers. But you cannot really test if a system reboots universally. Say that a system does not have the `shutdown` command, then the only thing you can test is to check if the hook triggers. It would be the user/sysadmin/distro responsibility to configure the conf file accordingly.

Said that, it's not a big deal imho since we are not testing the capability of `shutdown -r` of rebooting the system.

> It's enough time to validate that a shutdown was scheduled and cancel it again

I think this would be the solution in most cases.